README.md

```diff
@@ -18,8 +18,32 @@ Java 17 is the JDK version for development.
 Initially Hibernate ORM v6.6 is the dependency version.
 
 ### Build from source
+The following gradle plugins are chosen for static code analysis:
 
-### Test
+#### Code style check
+
+[Spotless](https://github.com/diffplug/spotless/tree/main/plugin-gradle) is a general-purpose formatting plugin; [Palantir Java Format](https://github.com/palantir/palantir-java-format) is a Java-specific auto-formatting plugin integrated with Spotless.
+
+To check whether format violation exists, run:
+
+```console
+$ ./gradlew spotlessCheck
+```
+
+If format violation found during the previous step, you can run auto-formatting task as:
+
+```console
+$ ./gradlew spotlessApply
+```
+
+#### Code quality check
+
+[Error Prone](https://github.com/tbroyer/gradle-errorprone-plugin) gradle plugin is chosen for Java code qualify analysis during compiling phrase. [NullAway](https://github.com/uber/NullAway) is a Java NPE prevention gradle plugin integrated with Error Prone. [JSpecify](https://jspecify.dev) annotations are used to help NullAway detect potential NPEs.
+
+To check whether either plugin complains, simply run the compiling gradle task:
+```console
+$ ./gradlew compileJava
```

Comment:
Are you sure you don't have to run `check`?

Reply:
I am sure. Maybe `clean` is needed as a precondition but `compileJava` is enough. `check` will include this step plus running test.

But I updated to use `check` instead.

Reply:
No, you're right, we don't want to run the tests here.  If we emulate the driver, we should do something like:

```
-x test -x spotlessApply check testClasses javadoc jar
```

Reply:
I just realized `check` won't trigger `ErrorProne` (and `NullAway`) code quality checking for `ErrorProne` is `compileJava` task, which goes beyond the scope of `check`, as shown by the below `dry-run` gradle output:
```
$ ./gradlew -x test clean check --dry-run
:clean SKIPPED
:spotlessInternalRegisterDependencies SKIPPED
:spotlessJava SKIPPED
:spotlessJavaCheck SKIPPED
:spotlessCheck SKIPPED
:check SKIPPED
```
I added a bad case of `NullAway` to test and found the above `check` step won't complain.
Definitively we don't need to specify `-x spotlessApply` for it won't take effect unless it is run manually by end-user. There might be some reason to specify it in java driver project, but I am sure we don't need to exclude it.

I also need to udpate the `static check` evergreen command to include `compileJava` as following:
```
/gradlew -PxmlReports.enabled=true --info -x test clean check compileJava
```

Reply:
We added `-x spotlessApply` for a good reason in the driver.  We had a release fail due to spotlessApply dirtying the tree, and realized that we hadn't notice it because our static checks were running `spotlessApply` before `spotlessCheck`, fixing the errors before they could be detected.

Reply:
I'm surprised that `check` doesn't depend on `compileJava`.  That seems odd.

Reply:
Traditionally check happens after compilation and usually static analysis is done on bytecode. Only Checker, ErrorProne and Lombok could do magic during compiling phase. They are exceptional, not the norm. Reportedly ErrorProne could lower false positive rate dramatically due to the fact that it has access to source code AST.